mariner/printer.py

```diff
@@ -139,7 +140,11 @@ def start_printing(self, filename: str) -> None:
             timeout_secs=2.0,
         )
         if "ok" not in response:
-            raise UnexpectedPrinterResponse(response)
+            raise UnexpectedPrinterResponse(response)   
+        timestamp_file = config.get_cache_directory() + '/timestamp.txt'
+        f = open(timestamp_file,"w")
+        f.write(str(int(time.time())))
+        f.close()        
```

Comment:
This feels a bit hacky 😅 You are basically doing this so you can later on figure out when the printing was started, right?

Here's some thoughts I have on the current implementation:

1. We will run into issues with prints which were started from other sources (e.g. from the printer itself).
2. This is doing a write to the filesystem from `mariner.printer`, which is meant really to just interface with the printer. The removal of the state file is happening from `mariner.server.api` - so it seems like we're mixing the roles of each module.
3. This will create all sorts of bugs with things like: if we start a print from the UI and then never come back to it, the state file won't get deleted. If we start a print from the printer and open the UI, we won't have a state file.

I feel like if we want to build this functionality correctly, we should create some sort of internal in-memory state for the server. Probably a second thread which is responsible for doing all communication with the printer *and* also responsible for polling the printer for its status - so that if a print is started from the printer we will get to know about it. This is something which I have been interested in for a while but haven't had the time to build: see #75 if you're interested

Reply:
Very valid points.

BTW, congrats on getting it back to a building state - I have no idea what you did, but I would like to know.

I need to have a chat with you privately about this and a few other things I have in mind, however I have no way of getting in touch with you. Could you please drop me an email on fcollingwood@gmail.com